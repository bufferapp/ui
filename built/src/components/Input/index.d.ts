export { default } from './InputWithRef';
//# sourceMappingURL=index.d.ts.map