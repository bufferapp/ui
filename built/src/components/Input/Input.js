import React from 'react';
import PropTypes from 'prop-types';
import * as Styles from './style';
import { Warning } from '../Icon';
import Text from '../Text';
export default class Input extends React.Component {
    render() {
        const { disabled, hasError, help, label, maxLength, id, name, onChange, onBlur, onKeyUp, prefix, placeholder, size, type, value, forwardedRef, required, icon, } = this.props;
        return (React.createElement(Styles.InputWrapper, null,
            label.length > 0 && (React.createElement(Text, { htmlFor: id, type: "label" }, label)),
            React.createElement(Styles.InputFieldWrapper, { prefix: prefix, size: size },
                icon && !prefix && React.createElement(Styles.StyledIcon, null, icon),
                React.createElement(Styles.InputStyled, { disabled: disabled, hasError: hasError, maxLength: maxLength, id: id, name: name, onChange: onChange, onBlur: onBlur, onKeyUp: onKeyUp, prefix: prefix, placeholder: placeholder, type: type, size: size, value: value, ref: forwardedRef, required: required, "aria-required": required ? true : undefined, icon: icon })),
            help.length > 0 && (React.createElement(Styles.HelpTextWrapper, null,
                hasError && React.createElement(Warning, { size: "medium" }),
                React.createElement(Styles.HelpText, { type: "help", htmlFor: name, hasError: hasError }, help)))));
    }
}
Input.propTypes = {
    disabled: PropTypes.bool,
    hasError: PropTypes.bool,
    required: PropTypes.bool,
    help: PropTypes.string,
    id: PropTypes.string,
    label: PropTypes.string,
    maxLength: PropTypes.string,
    name: PropTypes.string.isRequired,
    placeholder: PropTypes.string,
    prefix: PropTypes.shape({
        text: PropTypes.string.isRequired,
        paddingLeft: PropTypes.string.isRequired,
    }),
    onChange: PropTypes.func.isRequired,
    onBlur: PropTypes.func,
    onKeyUp: PropTypes.func,
    size: PropTypes.string,
    type: PropTypes.string,
    value: PropTypes.string,
    icon: PropTypes.node,
    forwardedRef: PropTypes.shape({ current: PropTypes.any }),
};
Input.defaultProps = {
    disabled: false,
    hasError: false,
    required: false,
    help: '',
    id: '',
    label: '',
    placeholder: '',
    size: 'regular',
    type: 'text',
    value: undefined,
    onBlur: () => { },
    onKeyUp: () => { },
    forwardedRef: undefined,
    prefix: null,
    maxLength: undefined,
    icon: undefined,
};
//# sourceMappingURL=Input.js.map