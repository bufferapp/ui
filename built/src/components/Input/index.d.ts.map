{"version":3,"file":"index.d.ts","sourceRoot":"","sources":["../../../../src/components/Input/index.ts"],"names":[],"mappings":"AAAA,OAAO,EAAE,OAAO,EAAE,MAAM,gBAAgB,CAAA"}