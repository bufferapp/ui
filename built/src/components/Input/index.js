export { default } from './InputWithRef';
//# sourceMappingURL=index.js.map