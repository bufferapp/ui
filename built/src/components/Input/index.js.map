{"version":3,"file":"index.js","sourceRoot":"","sources":["../../../../src/components/Input/index.ts"],"names":[],"mappings":"AAAA,OAAO,EAAE,OAAO,EAAE,MAAM,gBAAgB,CAAA","sourcesContent":["export { default } from './InputWithRef'\n"]}