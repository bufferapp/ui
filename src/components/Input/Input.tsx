import React from 'react'

import PropTypes from 'prop-types'
import * as Styles from './style'
import { Warning } from '../Icon'
import Text from '../Text'

export default class Input extends React.Component {
  render() {
    const {
      disabled,
      hasError,
      help,
      label,
      maxLength,
      id,
      name,
      onChange,
      onBlur,
      onKeyUp,
      prefix,
      placeholder,
      size,
      type,
      value,
      forwardedRef,
      required,
      icon,
    } = this.props
    return (
      <Styles.InputWrapper>
        {label.length > 0 && (
          <Text htmlFor={id} type="label">
            {label}
          </Text>
        )}
        <Styles.InputFieldWrapper prefix={prefix} size={size}>
          {icon && !prefix && <Styles.StyledIcon>{icon}</Styles.StyledIcon>}
          <Styles.InputStyled
            disabled={disabled}
            hasError={hasError}
            maxLength={maxLength}
            id={id}
            name={name}
            onChange={onChange}
            onBlur={onBlur}
            onKeyUp={onKeyUp}
            prefix={prefix}
            placeholder={placeholder}
            type={type}
            size={size}
            value={value}
            ref={forwardedRef}
            required={required}
            aria-required={required ? true : undefined}
            icon={icon}
          />
        </Styles.InputFieldWrapper>
        {help.length > 0 && (
          <Styles.HelpTextWrapper>
            {hasError && <Warning size="medium" />}
            <Styles.HelpText type="help" htmlFor={name} hasError={hasError}>
              {help}
            </Styles.HelpText>
          </Styles.HelpTextWrapper>
        )}
      </Styles.InputWrapper>
    )
  }
}

Input.propTypes = {
  /** It disables the input field. </i> */
  disabled: PropTypes.bool,
  /** It colors the field in red. */
  hasError: PropTypes.bool,
  /** Marks the input as required */
  required: PropTypes.bool,
  /** It adds an help text below the input box. */
  help: PropTypes.string,
  /** Id to link label with input for a11y */
  id: PropTypes.string,
  /** It adds a label on top of the input box. Make sure you also add an id for a11y */
  label: PropTypes.string,
  /** It adds a maxlength option for the input. */
  maxLength: PropTypes.string,
  /** It's the name of the input. */
  name: PropTypes.string.isRequired,
  /** It's the placeholder value of the input. */
  placeholder: PropTypes.string,
  /** Optional object describing fixed text that is placed inside the input, format is `{ text: '@', paddingLeft: '30px' }` */
  prefix: PropTypes.shape({
    text: PropTypes.string.isRequired,
    paddingLeft: PropTypes.string.isRequired,
  }),
  /** The onChange event */
  onChange: PropTypes.func.isRequired,
  /** The onBlur event */
  onBlur: PropTypes.func,
  /** The onKeyUp event */
  onKeyUp: PropTypes.func,
  /** This is the vertical size of the input field, can be `small`, `regular`, or `tall` */
  size: PropTypes.string,
  /** The type of the input */
  type: PropTypes.string,
  /** The value of the input */
  value: PropTypes.string,
  /** The value of the icon */
  icon: PropTypes.node,
  /**
   * this consumed by the default export that is wrapping the component into a ForwardRef
   * @ignore
   */
  forwardedRef: PropTypes.shape({ current: PropTypes.any }),
}

Input.defaultProps = {
  disabled: false,
  hasError: false,
  required: false,
  help: '',
  id: '',
  label: '',
  placeholder: '',
  size: 'regular',
  type: 'text',
  value: undefined,
  onBlur: () => {},
  onKeyUp: () => {},
  forwardedRef: undefined,
  prefix: null,
  maxLength: undefined,
  icon: undefined,
}
