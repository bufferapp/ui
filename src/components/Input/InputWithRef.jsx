import React from 'react'
import Input from './Input'

// This is a bit ugly but we need it otherwise automatic snapshots tests will fail
export default React.forwardRef((props, ref) => (
  <Input {...props} forwardedRef={ref} />
))
