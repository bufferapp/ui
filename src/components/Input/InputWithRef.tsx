import React from 'react'
import Input from './Input'

// This is a bit ugly but we need it otherwise automatic snapshots tests will fail
export default React.forwardRef((props, ref) => (
  // @ts-expect-error TS(2769) FIXME: No overload matches this call.
  <Input {...props} forwardedRef={ref} />
))
