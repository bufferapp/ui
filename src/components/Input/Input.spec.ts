/* eslint-disable react/jsx-filename-extension */
import snap from 'jest-auto-snapshots'
import 'jest-styled-components'

import Input from './Input'

snap(Input, './Input.jsx')
