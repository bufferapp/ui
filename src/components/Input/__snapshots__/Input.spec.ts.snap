// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`jest-auto-snapshots > Input Matches snapshot when boolean prop "disabled" is set to: "false" 1`] = `
<styled.div>
  <Text
    color="grayDarker"
    hasError={false}
    htmlFor="jest-auto-snapshots String Fixture"
    type="label"
  >
    jest-auto-snapshots String Fixture
  </Text>
  <styled.div
    prefix={
      Object {
        "paddingLeft": "jest-auto-snapshots String Fixture",
        "text": "jest-auto-snapshots String Fixture",
      }
    }
    size="jest-auto-snapshots String Fixture"
  >
    <styled.input
      aria-required={true}
      disabled={false}
      hasError={true}
      icon={<NodeFixture />}
      id="jest-auto-snapshots String Fixture"
      maxLength="jest-auto-snapshots String Fixture"
      name="jest-auto-snapshots String Fixture"
      onBlur={[MockFunction]}
      onChange={[MockFunction]}
      onKeyUp={[MockFunction]}
      placeholder="jest-auto-snapshots String Fixture"
      prefix={
        Object {
          "paddingLeft": "jest-auto-snapshots String Fixture",
          "text": "jest-auto-snapshots String Fixture",
        }
      }
      required={true}
      size="jest-auto-snapshots String Fixture"
      type="jest-auto-snapshots String Fixture"
      value="jest-auto-snapshots String Fixture"
    />
  </styled.div>
  <styled.div>
    <WarningIcon
      size="medium"
    />
    <Styled(Text)
      hasError={true}
      htmlFor="jest-auto-snapshots String Fixture"
      type="help"
    >
      jest-auto-snapshots String Fixture
    </Styled(Text)>
  </styled.div>
</styled.div>
`;

exports[`jest-auto-snapshots > Input Matches snapshot when boolean prop "hasError" is set to: "false" 1`] = `
<styled.div>
  <Text
    color="grayDarker"
    hasError={false}
    htmlFor="jest-auto-snapshots String Fixture"
    type="label"
  >
    jest-auto-snapshots String Fixture
  </Text>
  <styled.div
    prefix={
      Object {
        "paddingLeft": "jest-auto-snapshots String Fixture",
        "text": "jest-auto-snapshots String Fixture",
      }
    }
    size="jest-auto-snapshots String Fixture"
  >
    <styled.input
      aria-required={true}
      disabled={true}
      hasError={false}
      icon={<NodeFixture />}
      id="jest-auto-snapshots String Fixture"
      maxLength="jest-auto-snapshots String Fixture"
      name="jest-auto-snapshots String Fixture"
      onBlur={[MockFunction]}
      onChange={[MockFunction]}
      onKeyUp={[MockFunction]}
      placeholder="jest-auto-snapshots String Fixture"
      prefix={
        Object {
          "paddingLeft": "jest-auto-snapshots String Fixture",
          "text": "jest-auto-snapshots String Fixture",
        }
      }
      required={true}
      size="jest-auto-snapshots String Fixture"
      type="jest-auto-snapshots String Fixture"
      value="jest-auto-snapshots String Fixture"
    />
  </styled.div>
  <styled.div>
    <Styled(Text)
      hasError={false}
      htmlFor="jest-auto-snapshots String Fixture"
      type="help"
    >
      jest-auto-snapshots String Fixture
    </Styled(Text)>
  </styled.div>
</styled.div>
`;

exports[`jest-auto-snapshots > Input Matches snapshot when boolean prop "required" is set to: "false" 1`] = `
<styled.div>
  <Text
    color="grayDarker"
    hasError={false}
    htmlFor="jest-auto-snapshots String Fixture"
    type="label"
  >
    jest-auto-snapshots String Fixture
  </Text>
  <styled.div
    prefix={
      Object {
        "paddingLeft": "jest-auto-snapshots String Fixture",
        "text": "jest-auto-snapshots String Fixture",
      }
    }
    size="jest-auto-snapshots String Fixture"
  >
    <styled.input
      disabled={true}
      hasError={true}
      icon={<NodeFixture />}
      id="jest-auto-snapshots String Fixture"
      maxLength="jest-auto-snapshots String Fixture"
      name="jest-auto-snapshots String Fixture"
      onBlur={[MockFunction]}
      onChange={[MockFunction]}
      onKeyUp={[MockFunction]}
      placeholder="jest-auto-snapshots String Fixture"
      prefix={
        Object {
          "paddingLeft": "jest-auto-snapshots String Fixture",
          "text": "jest-auto-snapshots String Fixture",
        }
      }
      required={false}
      size="jest-auto-snapshots String Fixture"
      type="jest-auto-snapshots String Fixture"
      value="jest-auto-snapshots String Fixture"
    />
  </styled.div>
  <styled.div>
    <WarningIcon
      size="medium"
    />
    <Styled(Text)
      hasError={true}
      htmlFor="jest-auto-snapshots String Fixture"
      type="help"
    >
      jest-auto-snapshots String Fixture
    </Styled(Text)>
  </styled.div>
</styled.div>
`;

exports[`jest-auto-snapshots > Input Matches snapshot when passed all props 1`] = `
<styled.div>
  <Text
    color="grayDarker"
    hasError={false}
    htmlFor="jest-auto-snapshots String Fixture"
    type="label"
  >
    jest-auto-snapshots String Fixture
  </Text>
  <styled.div
    prefix={
      Object {
        "paddingLeft": "jest-auto-snapshots String Fixture",
        "text": "jest-auto-snapshots String Fixture",
      }
    }
    size="jest-auto-snapshots String Fixture"
  >
    <styled.input
      aria-required={true}
      disabled={true}
      hasError={true}
      icon={<NodeFixture />}
      id="jest-auto-snapshots String Fixture"
      maxLength="jest-auto-snapshots String Fixture"
      name="jest-auto-snapshots String Fixture"
      onBlur={[MockFunction]}
      onChange={[MockFunction]}
      onKeyUp={[MockFunction]}
      placeholder="jest-auto-snapshots String Fixture"
      prefix={
        Object {
          "paddingLeft": "jest-auto-snapshots String Fixture",
          "text": "jest-auto-snapshots String Fixture",
        }
      }
      required={true}
      size="jest-auto-snapshots String Fixture"
      type="jest-auto-snapshots String Fixture"
      value="jest-auto-snapshots String Fixture"
    />
  </styled.div>
  <styled.div>
    <WarningIcon
      size="medium"
    />
    <Styled(Text)
      hasError={true}
      htmlFor="jest-auto-snapshots String Fixture"
      type="help"
    >
      jest-auto-snapshots String Fixture
    </Styled(Text)>
  </styled.div>
</styled.div>
`;

exports[`jest-auto-snapshots > Input Matches snapshot when passed only required props 1`] = `
<styled.div>
  <styled.div
    prefix={null}
    size="regular"
  >
    <styled.input
      disabled={false}
      hasError={false}
      id=""
      name="jest-auto-snapshots String Fixture"
      onBlur={[Function]}
      onChange={[MockFunction]}
      onKeyUp={[Function]}
      placeholder=""
      prefix={null}
      required={false}
      size="regular"
      type="text"
    />
  </styled.div>
</styled.div>
`;
